vllm/worker/model_runner.py

```diff
@@ -666,6 +667,60 @@ def list_loras(self) -> Set[int]:
             raise RuntimeError("LoRA is not enabled.")
         return self.lora_manager.list_loras()
 
+    @torch.inference_mode()
+    def compile_model(self, kv_caches: List[KVCache]) -> None:
+        assert not self.model_config.enforce_eager
+        logger.info("Compiling the model using torch.compile, which may take "
+                    "several minutes. If you prefer to skip the compilation, "
+                    "albeit with performance degradation, set "
+                    "'enforce_eager=True' or use '--enforce-eager' in CLI.")
+
+        start_time = time.perf_counter()
+        self.compiled_model = torch.compile(self.model, fullgraph=True)
```

Comment:
how will it work with world_size > 1?
will the compiled graph contain breaks to perform collective communication operations?

Reply:
I haven't added functional collectives yet to the PR, but Pytorch has support for those. This being said, updating to Pytorch 2.2 would simplify the implementation of these significantly